### PR TITLE
Add disk buffering config options to RUM conf

### PR DESCRIPTION
**Pull request by mhennoch**
Currently iOS and android have different config options for enabling and configuring disk buffering.

Android
enableDiskBuffering
limitDiskUsageMegabytes

iOS
enableDiskCache
spanDiskCacheMaxSize

Also react native needs to use the same conf. After discussions it seems "buffering" makes more sense so let's use android version. Not sure where to put mobile only conf options put them into the same table.

### CODE REVIEW

**Comment by mhennoch** [comment]
I have read the CLA Document and I hereby sign the CLA

File: specification/configuration.md [hunk 1]
```diff
@@ -266,14 +266,16 @@ etc.).
...
+| `applicationName` ()           | Instrumented application name. [1]                                                                                                                                                                         |
+| `globalAttributes` ()          | OpenTelemetry [Attributes](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/common/README.md#attribute) that will be added to every span produced by the RUM library. |
+| `deploymentEnvironment` ()     | Sets the environment (`deployment.environment` span attribute) for all spans.                                                                                                                              |
+| `enableDiskBuffering` (false)  | **Mobile only.** Enables the storage-based buffering of telemetry                                                                                                                                          |
```
Comment (breedx-splk) [inline specification/configuration.md:277]:
```suggestion
| `enableDiskBuffering` (false)  | **Mobile only.** Enables the storage-based buffering of telemetry.|
```

**Review by breedx-splk — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -9,6 +9,7 @@
 - Require a CLA Assistant GitHub workflow. (#269)
 - Update the CLA notice in `CONTRIBUTING.md` template. (#269, #274)
 - Add Renovate as an acceptable alternative to Dependabot. (#271)
+- Add disk buffering configuration options for RUM (#275)
```
Comment (pellared) [inline CHANGELOG.md:12]:
```suggestion
- Add disk buffering configuration options for RUM mobile instrumentation libraries. (#275)
```

**Review by pellared — APPROVED** [review]

File: specification/configuration.md (hunk 1 above)
Comment (pellared) [reply specification/configuration.md:277]:
```suggestion
| `enableDiskBuffering` (false)  | **Mobile only.** Enables the storage-based buffering of telemetry.                                                                                                                                         |
```

**Comment by Kielek** [comment]
@signalf/gdi-ios-maintainers, FYI

**Comment by Kielek** [comment]
@signalfx/gdi-ios-maintainers, FYI

**Review by Kielek — APPROVED** [review]

**Pull request merged**